lib/valueflow.cpp

```diff
@@ -3997,9 +3997,9 @@ static std::list<ValueFlow::Value> truncateValues(std::list<ValueFlow::Value> va
         if (value.isIntValue() && sz > 0 && sz < 8) {
             const MathLib::biguint unsignedMaxValue = (1ULL << (sz * 8)) - 1ULL;
             const MathLib::biguint signBit = 1ULL << (sz * 8 - 1);
-            value.intvalue &= unsignedMaxValue;
+            value.intvalue &= static_cast<long long>(unsignedMaxValue);
```

Comment:
I do not see a safety improvement here.. just a cast. I question that it improves the readability. And it is dangerous to use casts it hides sanitizer/compiler warnings ...

Reply:
This will be done implicitly anyways - I just doing it explicitly so UBSAN know it is actually intended.

Reply:
about conversion warnings.. if there are bugs I rather fix the bugs. If there are not bugs I do not want to add casts because in the long run those could hide real bugs. I think that rewriting the code can be an option sometimes, unsigned types should be avoided.

I guess that in this case the unsignedMaxValue does not need to be unsigned. We know that the value of `sz` is 1-7.